evaluator/Updaters/global_price_updater.py

```diff
@@ -145,13 +145,12 @@ def _set_symbol_time_frame_next_update_time(self, symbol, time_frame, candle_tim
         if time_frame not in self.time_frame_next_update:
             self.time_frame_next_update[time_frame] = {}
 
-        if symbol not in self.time_frame_next_update[time_frame] or not self.time_frame_next_update[time_frame][symbol]:
-            if candle_time:
-                self.time_frame_next_update[time_frame][symbol] = candle_time
-            else:
-                self.time_frame_next_update[time_frame][symbol] = time.time()
+        if candle_time:
+            self.time_frame_next_update[time_frame][symbol] = candle_time
+        elif symbol not in self.time_frame_next_update[time_frame]:
+            self.time_frame_next_update[time_frame][symbol] = time.time()
 
-        self.time_frame_next_update[time_frame][symbol] += TimeFramesMinutes[time_frame] * MINUTE_TO_SECONDS
+        self.time_frame_next_update[time_frame][symbol] += TimeFramesMinutes[time_frame] * 2 * MINUTE_TO_SECONDS
```

Comment:
Why 2x ?

Reply:
Because PriceIndexes.IND_PRICE_TIME.value is the previous candle open time.
So we have to wait for the previous candle close = TimeFramesMinutes[time_frame] * MINUTE_TO_SECONDS and the current candle to close = TimeFramesMinutes[time_frame] * MINUTE_TO_SECONDS.
--> Wait for TimeFramesMinutes[time_frame] * 2 * MINUTE_TO_SECONDS s